job_board/views.py

```diff
@@ -23,7 +23,7 @@ def get(self, request, *args, **kwargs):
     def post(self, request, *args, **kwargs):
         form = self.form_class(request.POST)
         if form.is_valid():
-            posts = SearchEngine.search(self, form.save(commit=False))
+            posts = SearchEngine.search(form.save(commit=False))
```

Comment:
@DeanBiton Do'nt you think this part should have client test also
There are returned values here, and you could verify it I is ubreakable, WDYT?

Reply:
> @DeanBiton Do'nt you think this part should have client test also There are returned values here, and you could verify it I is ubreakable, WDYT?

@Yarboa The view still works the same as before and I think the tests in 
https://github.com/redhat-beyond/JobSeeker/blob/main/job_board/tests/test_search_view.py suffice.

Also, the only thing that could raise an exception from the search function is if the object sent is not from the Preference model which will not happen in our case because we check this beforehand.
Overall, I could add a check for this inside the search function and return -1 in case it's not the expected object type.